File: CodeForAndroid/ContactTest/app/src/main/java/com/example/contacttest/MainActivity.java
package com.example.contacttest;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.Bitmap;
import android.net.Uri;
import android.nfc.Tag;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.provider.MediaStore;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.alibaba.fastjson.JSONObject;
import com.example.contacttest.bean.Bean.NewMessage;
import com.example.contacttest.bean.UserOrdinary;

import org.json.JSONException;

import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.net.Socket;
import java.net.UnknownHostException;
import java.nio.charset.StandardCharsets;
import java.util.Arrays;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.LinkedBlockingQueue;
import java.util.concurrent.ThreadPoolExecutor;
import java.util.concurrent.TimeUnit;

public class MainActivity extends AppCompatActivity {

    static boolean logflag;

    private final static String TAG = "Connect";

    public final static ExecutorService threadPoolExecutor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 3 + 2);

    Button btn1;
    TextView text1,text2;
    EditText editText1,editText2;
    ImageButton image1;


    private static String ip = "120.79.87.21";
    private static int portUpload = 5423;
    private static int port = 5422;

    Connect ct;

    Bitmap bitmap;

    public void toastText(String message){
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
    }

    public void handleException(Exception e, String prefix)
    {
        e.printStackTrace();
        toastText(prefix + e.toString());
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ct = Connect.getConncet();
        Connect.Init(this);
        btn1 = findViewById(R.id.btn1);
        text1 = findViewById(R.id.text1);
        text2 = findViewById(R.id.text2);
        editText1 = findViewById(R.id.edit1);
        editText2 = findViewById(R.id.edit2);
        image1 = findViewById(R.id.image1);
        image1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent();
                intent.setAction(Intent.ACTION_PICK);
                intent.setType("image/*");
                startActivityForResult(intent, 2);

            }
        });
        btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                threadPoolExecutor.execute(testconect);
            }
        });

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data){
        if (requestCode == 2) {
            // 从相册返回的数据
            if (data != null) {
                // 得到图片的全路径
                Uri uri = data.getData();
                try {
                    bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), uri);
                    image1.setImageBitmap(bitmap);
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    @SuppressLint("HandlerLeak")
    Handler handler = new Handler(){
        @Override
        public void handleMessage(Message msg) {
            super.handleMessage(msg);
            Bundle data = msg.getData();
            String val = data.getString("value");
            text2.setText(val);
            Log.d(TAG,"请求结果-->" + val);
        }
    };

    Runnable testconect = new Runnable() {
        @Override
        public void run() {
            SharedPreferences sp_user = getSharedPreferences("user", Context.MODE_PRIVATE);
            SharedPreferences.Editor editor_user = sp_user.edit();
            editor_user.putInt("PhoneNum", 123);
            editor_user.apply();

            bitmap = ct.getHeadPortrait();
            image1.setImageBitmap(bitmap);
        }
    };

}

File: PicPic/app/src/main/java/com/xz2dai/picpic/MainActivity.kt
package com.xz2dai.picpic

import PixivDownloader
import android.graphics.Bitmap
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.concurrent.thread


class MainActivity : AppCompatActivity(),View.OnClickListener  {

    val pixivDownloader:PixivDownloader = PixivDownloader()
    var reSize:Boolean = false



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_getIMG.setOnClickListener(this)
        btn_changeLocal.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when(v){
            btn_getIMG -> getImg()
            btn_changeLocal -> changeLocal()
            imageview ->{
                if (!reSize){
                    //初始处于缩放状态，点击后放大到全屏
                }else{
                    //处于放大状态，缩放图片
                }

            }
        }
    }

    private fun changeLocal() {

    }

    fun getImg(){
        var pvid:String
        var bitmap:Bitmap
        if(edittext.text!=null){
            pvid = edittext.text.toString()
            thread {
                bitmap = pixivDownloader.GetIMG(pvid)       //获取图片
                runOnUiThread(Runnable {        //修改View需要在创建view的线程中进行
                    imageview.setImageBitmap(bitmap)
                })
            }
        }else{
            Toast.makeText(this, "please input a pvid", Toast.LENGTH_SHORT).show()
        }
    }

}